### PR TITLE
DevTest: Formspec tests, children getter, better lighttool

### PR DESCRIPTION
**Pull request by Wuzzy2**
It's a minor improvement for DevTest:

* Adds more formspec tests for sound styles and model[]
* Adds a "children getter" tool. This shows the list of objects attached to another object (using `get_children`)
* Improvement of the light tool: With the "place" key, you check the light level of the node that you touched. Also displays the pos of the node you just checked. Tooltip was also improved.

To test the children getter, just attach objects to each other, then use the children getter on them.

### CODE REVIEW

**Comment by Wuzzy2** [comment]
I've added the "Node Meta Editor" to DevTest, this is a tool to edit and view node metadata.

File: games/devtest/mods/testtools/init.lua [hunk 1]
```diff
@@ -666,6 +695,63 @@ local function use_loadstring(param, player)
...
+	end
+	minetest.show_formspec(user:get_player_name(),
+		"testtools:node_meta_editor",
+		"size[15,9]"..
+		"label[0,0;"..F(S("Current keys:")).."]"..
+		textlist..
+		"field[3,0.5;12,1;key;"..F(S("Key"))..";"..F(key).."]"..
+		"textarea[3,1.5;12,6;value;"..F(S("Value"))..";"..F(value).."]"..
+		"button[0,8;3,1;get;"..F(S("Get value")).."]"..
+		"button[4,8;3,1;set;"..F(S("Set value")).."]"..
```
Comment (SmallJoker) [inline games/devtest/mods/testtools/init.lua:717]: How about removing metadata fields?

**Comment by Wuzzy2** [comment]
> How about removing metadata fields?

This is done by clearing the value field and then hitting "Set value".

File: games/devtest/mods/testtools/init.lua [hunk 2]
```diff
@@ -634,6 +613,56 @@ minetest.register_tool("testtools:object_attacher", {
...
+				selected_object = pointed_thing.ref
+			elseif pointed_thing.type == "nothing" then
+				selected_object = user
+			else
+				return
+			end
+			if selected_object:is_player() then
+				self_name = "player '"..selected_object:get_player_name().."'"
+			elseif selected_object:get_luaentity() then
+				self_name = "LuaEntity '"..selected_object:get_entity_name().."'"
```
Comment (SmallJoker) [inline games/devtest/mods/testtools/init.lua:638]: `get_entity_name` is marked for deprecation for years. `selected_object:get_luaentity().name` seems to be the new/correct way.
Related discussion: https://irc.minetest.net/minetest-dev/2021-02-27#i_5790818

EDIT: Also this code is the same as in the loop below (except for the `"* "` prefix). A function would avoid this duplication, but the decision for this is up to you.

File: games/devtest/mods/testtools/init.lua [hunk 3]
```diff
@@ -666,6 +695,63 @@ local function use_loadstring(param, player)
...
+	local keys = {}
+	local mtable = meta:to_table()
+	for k,_ in pairs(mtable.fields) do
+		table.insert(keys, k)
+	end
+	if playername then
+		node_meta_latest_keylist[playername] = keys
+	end
+	local keylist = table.concat(keys, ",")
+	return keylist
```
Comment (SmallJoker) [inline games/devtest/mods/testtools/init.lua:731]: 1. Useless temporary variable
2. Needs formspec escaping before concat

File: games/devtest/mods/testtools/init.lua [hunk 4]
```diff
@@ -666,6 +695,63 @@ local function use_loadstring(param, player)
...
+	else
+		textlist = ""
+	end
+	minetest.show_formspec(user:get_player_name(),
+		"testtools:node_meta_editor",
+		"size[15,9]"..
+		"label[0,0;"..F(S("Current keys:")).."]"..
+		textlist..
+		"field[3,0.5;12,1;key;"..F(S("Key"))..";"..F(key).."]"..
+		"textarea[3,1.5;12,6;value;"..F(S("Value"))..";"..F(value).."]"..
```
Comment (SmallJoker) [inline games/devtest/mods/testtools/init.lua:715]:
```suggestion
		"textarea[3,1.5;12,6;value;"..F(S("Value (use empty value to delete key)"))..";"..F(value).."]"..
```

File: games/devtest/mods/testtools/init.lua [hunk 5]
```diff
@@ -728,5 +814,66 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
...
+				local meta = minetest.get_meta(pos)
+				if not keylist_table then
+					return
+				end
+				if #keylist_table == 0 then
+					return
+				end
+				local key = keylist_table[evnt.index]
+				local value = meta:get_string(key)
+				local keylist = table.concat(keylist_table, ",")
```
Comment (SmallJoker) [inline games/devtest/mods/testtools/init.lua:851]: Needs formspec escaping before concat

**Comment by Wuzzy2** [comment]
All requested updates done!

**Review by SmallJoker — APPROVED** [review]
Works (quick test)

**Pull request merged**